File: Data Structures & Algorithms/three-integer-sum/submission-9.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    threeSum(nums) {
        
        const n = nums.length;

        const sorted = [...nums].sort((a, b) => a - b);

        const set = new Set();

        for(let i = 0; i < n - 2; i += 1) {
            for(let j = i + 1; j < n - 1; j += 1) {
                for(let k = j + 1; k < n; k += 1) {
                    
                    if(sorted[i] + sorted[j] + sorted[k] === 0) {
                        // set.add(`${[sorted[i], sorted[j], sorted[k]]}`);
                        set.add(JSON.stringify([sorted[i], sorted[j], sorted[k]]));
                    }

                }
            }
        }

        // return Array.from(set).map((triplet) => triplet.split(",").map(Number));
        return Array.from(set).map((item) => JSON.parse(item));

    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-9.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        n = len(nums)

        res = [1] * n

        i = 1
        while i < n:
            res[i] = res[i - 1] * nums[i - 1]
            i += 1
        
        prod = 1
        i = n - 1
        while i >= 0:
            prod = prod * (1 if i + 1 >= n else nums[i + 1])
            res[i] = res[i] * prod
            i -= 1
        
        return res
        

